### PR TITLE
Adding Note into Parameters

### LINKED ISSUE #196: Phone Number Bug: Not restrictive enough
Current implementation for regex is to check if length >= 3 and all characters is number.

This allows phone number shorter than 8 character to be entered:
Screenshot 2023-11-03 at 1 01 01 PM

As well as phone number with more than 8 characters:
Screenshot 2023-11-03 at 1 02 08 PM

Let's restrict it to the Singapore context
- First number must be either 6, 8, 9
- Length of number must be exactly 8

### LINKED ISSUE #212: [PE-D][Tester C] UG flaw regarding invalid inputs/ arguments [UG][edit_p][Constraints on arguments]
No mention about invalid inputs and argument requirements in UG, feedback is only gotten from the result display box. 
Lack of information on the consequences of using blanks

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: sk2001git/ped#5

### LINKED ISSUE #219: [PE-D][Tester C] Space in tags [Feature Flaw][Tabs/Space in argument field]
We allow space in tags just before the first letter due to trim(). However, any tabs after that doesnt work and I believe this should be specified in either the result display or UG 

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: sk2001git/ped#7

### LINKED ISSUE #246: Lack of specification on Tag's Restriction on spaces
Spaces is not allowed in tag, but we didn't specify it like

"add_tag_d 2 t\Clinical Pathology" will incur error

but 

"add_tag_d 2 t\ClinicalPathology" is preferred

### PR DESCRIPTION
**Pull request by kimshitong**
Closes #196 
Closes #219 
Closes #212 
Closes #246

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -166,13 +166,17 @@ The features are broken down to into:
...
 The following table gives an overview of what each parameter used in the Features section is for.
 
-| Parameter    | Description                                                          |
-|--------------|----------------------------------------------------------------------|
-| NAME         | The full name of the doctor/patient                                  |
-| NRIC         | The National Registration Identity Card number of the doctor/patient |
-| PHONE_NUMBER | The phone number of the doctor/patient                               |
-| INDEX        | The position of the appointment/doctor/patient in the filtered list  |
-
+| Parameter    | Description                                                          | Valid Input Note                                                                                                |
```
Comment (cheeggered) [inline docs/UserGuide.md:169]: Perhaps use another term instead of "Note", ie. Valid Parameter Input

File: docs/UserGuide.md [hunk 2]
```diff
@@ -166,13 +166,17 @@ The features are broken down to into:
...
+|--------------|----------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|
+| NAME         | The full name of the doctor/patient                                  | Only accepts and start with an alphanumeric character                                                           |
+| NRIC         | The National Registration Identity Card number of the doctor/patient | Starts with S/T/F/G/M and followed by 7 digits and end with a uppercase alphabet                                |
+| PHONE_NUMBER | The phone number of the doctor/patient                               | Only contain numbers, and it should be at least 3 digits long. The max-length of phone number is not restricted |
```
Comment (cheeggered) [inline docs/UserGuide.md:173]: No need to include "The max-length of phone number is not restricted"

File: docs/UserGuide.md [hunk 3]
```diff
@@ -166,13 +166,17 @@ The features are broken down to into:
...
+| INDEX        | The position of the appointment/doctor/patient in the filtered list  | Must be a positive integer (1, 2, 3, ...)                                                                       |
+| TAG          | The tag of the appointment/doctor/patient                            | Only accepts and start with alphanumeric character without spacing.                                             |
+
+<div markdown="span" class="alert alert-note">:information_source: <b>Note:</b>
+An "alphanumeric" character is a character that can be either a letter (A to Z, both uppercase and lowercase) or a number (0 to 9).
```
Comment (cheeggered) [inline docs/UserGuide.md:178]: Great inclusion of the definition of "alphanumeric", makes it much clearer for users!

File: docs/UserGuide.md (hunk 2 above)
Comment (kimshitong) [reply docs/UserGuide.md:173]: Done

File: docs/UserGuide.md (hunk 1 above)
Comment (kimshitong) [reply docs/UserGuide.md:169]: Ok, thanks

File: docs/UserGuide.md (hunk 3 above)
Comment (yanghengtang) [reply docs/UserGuide.md:178]: stfu

**Review by yanghengtang — APPROVED** [review]

**Pull request merged**